docs/source/notes/extending.rst

```diff
@@ -445,6 +445,64 @@ Also see the ``MetadataTensor`` example below for another variation on this
 pattern but instead always returns a ``MetadataTensor`` to propagate metadata
 through operations in the :mod:`torch` API.
 
+The ``__torch_function__`` protocol is designed for full coverage of the API,
+partial coverage may lead to undesirable results, in particular, certain
+functions raising a ``TypeError``. This is especially true for subclasses,
+where all three of `torch.add`, `torch.Tensor.__add__` and `torch.Tensor.add`
+must be covered, even if they return exactly the same result. Failing to do
+this may also lead to infinite recursion. If one requires the implementation
+of a function from ``torch.Tensor`` subclasses, they must use
```

Comment:
Maybe consider reword: If you have subclassed Tensor, and you want to call the ordinary PyTorch function on exactly this Tensor, you must use `super().__torch_function__` inside `__torch_function__` to invoke it. In particular, do not directly call `func`, as that will infinite loop."